### ISSUE TITLE
GradleBuildStatus.retry should backoff

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#83971]
Currently, it retries immediately

This probably works out most of the time because gradle does lots of things before getting to the point we want to retry anyway, but we should have some kind of pause before retrying.

/cc @blasten

**Comment by blasten** [comment]
right. http retries with exponential backoff is enabled by default: https://docs.gradle.org/current/userguide/dependency_resolution.html#sub:http-retries
